src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -69,12 +69,20 @@ async def _git_clone_repo(
     await run_cmd_line(cmd)
 
 
-async def _git_get_current_sha(directory: Path) -> str:
+async def _git_get_FETCH_HEAD_sha(directory: Path) -> str:
     cmd = ["git", "rev-parse", "--short", "FETCH_HEAD"]
     sha = await run_cmd_line(cmd, str(directory))
     return sha.strip("\n")
 
 
+async def _git_get_sha_of_tag(directory: Path, tag: str) -> str:
```

Comment:
minor: you could also have
```python
async def _git_get_sha(director:Path, tag: Optional[str] = None) -> str:
  cmd = ["git", "rev-parse", "--short", f"{tag}" if tag else "FETCH_HEAD"]
  sha = await run_cmd_line(cmd, str(directory))
  return sha.strip("\n")
```

Reply:
re: first: Here I get on wsl2
```
osparc-simcore on  master [$] via 
❯ git rev-parse --short v1.20.3
fatal: Needed a single revision
```

that's why I went for the other guy, just to be safe.